Mathlib/CategoryTheory/Adjunction/Over.lean

```diff
@@ -29,27 +30,76 @@ open Category Limits Comonad
 
 variable {C : Type u} [Category.{v} C] (X : C)
 
+
+namespace Over
+
+open Limits
+
+-- Porting note: removed semireducible from the simps config
+/-- Given a morphism `f : X ⟶ Y`, the functor `baseChange f` takes morphisms over `Y` to morphisms
+over `X` via pullbacks. -/
+@[simps! (config := { simpRhs := true}) obj_left obj_hom map_left]
+def baseChange [HasPullbacks C] {X Y : C} (f : X ⟶ Y) : Over Y ⥤ Over X where
+  obj g := Over.mk (pullback.snd : pullback g.hom f ⟶ _)
+  map i := Over.homMk (pullback.map _ _ _ _ i.left (𝟙 _) (𝟙 _) (by simp) (by simp))
+  map_id Z := by
+    apply Over.OverMorphism.ext; apply pullback.hom_ext
+    · dsimp; simp
+    · dsimp; simp
+  map_comp f g := by
+    apply Over.OverMorphism.ext; apply pullback.hom_ext
+    · dsimp; simp
+    · dsimp; simp
+#align category_theory.limits.base_change CategoryTheory.Over.baseChange
+
+-- deprecated on 2024-05-15
+@[deprecated] noncomputable alias Limits.baseChange := Over.baseChange
```

Comment:
```suggestion
@[deprecated (since := "2024-05-15")] noncomputable alias Limits.baseChange := Over.baseChange
```
Also note that since #13206 you can type `depr` + `TAB` and a code snippet
````
@[deprecated (since := "current date")] alias  := 
```
will appear